theme/blast_report.tpl.php

```diff
@@ -86,10 +86,41 @@
       </div>
     </div>
 
-    <br />
-    <div class="num-results">
-      <strong>Number of Results</strong>: <?php print $num_results; ?>
-    </div>
+    <?php
+      if (variable_get('blast_ui_cvitjs_enabled', false)
+            && isset($blast_job->blastdb->cvitjs_enabled)
+            && $blast_job->blastdb->cvitjs_enabled == '1') {
+        $cvitjs_location = variable_get('blast_ui_cvitjs_location', '');
+    ?>
+      <!-- CViTjs image of BLAST hits, if enabled -->
+      <div class="cvitjs">
+        <div id="title-div"></div>
+        <div id="cvit-div"></div>
+      </div>
+      <?php
+        drupal_add_js(array(
+          'blast_ui'=> array(
+            'dataset' => $blast_job->blastdb->db_name)
+          ),
+          'setting'
+        );
+        drupal_add_js(array(
+          'blast_ui'=> array(
+            'gff' => '../../' . $blast_job->files->result->gff)),'setting'
+        );    
+        $base = drupal_get_path('module','blast_ui') 
+                . DIRECTORY_SEPARATOR . $cvitjs_location
+                . DIRECTORY_SEPARATOR . 'js'
+                . DIRECTORY_SEPARATOR . 'lib' . DIRECTORY_SEPARATOR;
+        drupal_add_css($base.'bootstrap/css/bootstrap.min.css',array('preprocess'=>FALSE));
+        drupal_add_css($base.'hopscotch/css/hopscotch.min.css',array('preprocess'=>FALSE));
+        drupal_add_css($base.'../../css/cvit.css',array('preprocess'=> FALSE));
+        drupal_add_js($base.'require/require.js',array('group'=>'JS_LIBRARY','type'=>'file'));
+        drupal_add_js($base.'require/blast_ui-config.js',array('group'=>'JS_THEME'));
+      ?>
```

Comment:
Ideally, CViTjs would be added as a library rather then cloned within the blast module... an example showing how to do this is my [cvitembed module](https://github.com/UofS-Pulse-Binfo/cvitembed). Specifically, 
1. CViTjs is cloned into sites/all/libraries
2. [You tell the libraries API that its there](https://github.com/UofS-Pulse-Binfo/cvitembed/blob/master/cvitembed.module#L81)
3. When you need it (e.g. right here ;-) ), you add the CSS/JS as follows:
```php
$cvitjs_lib_path = libraries_get_path('cvitjs') . DIRECTORY_SEPARATOR;
drupal_add_css($cvitjs_lib_path . 'js/lib/bootstrap/css/bootstrap.min.css',array('preprocess'=>FALSE));
drupal_add_css($cvitjs_lib_path . 'js/lib/hopscotch/css/hopscotch.min.css',array('preprocess'=>FALSE));
drupal_add_css($cvitjs_lib_path . 'css/cvit.css',array('preprocess'=> FALSE));
drupal_add_js($cvitjs_lib_path . 'js/lib/require/require.js',array('group'=>'JS_LIBRARY','type'=>'file'));
drupal_add_js($cvitjs_lib_path . 'js/lib/require/blast_ui-config.js',array('group'=>'JS_THEME'));
```

Reply:
As long as this can pull from the CViTjs repository (which appears to be the case), I'll give this a try.

Reply:
We (i.e. Andrew W) will need to make some changes to CViTjs to better package it before attempting this, changes that may be extensive. Also, changes to the the admin page to remove some of the awkward hand work and will need some careful thought. I would like to leave this to a later pull request.